### ISSUE TITLE
GridFieldEditableColumns does not work with ManyManyThroughList

### ISSUE THREAD

**Issue by ntd** [symbiote/silverstripe-gridfieldextensions#309]
I just tried to substitute the two occurrences of `$list instanceof ManyManyList` with `($list instanceof ManyManyList || $list instanceof ManyManyThroughList)` but it does not seem to work properly.

Let's pretend to have the following data model:
```php
class Team extends DataObject
{
    private static $many_many = [
        "Supporters" => [
            'through' => TeamSupporter::class,
            'from' => 'Team',
            'to' => 'Supporter',
        ]
    ];
}
class Supporter extends DataObject
{
    private static $belongs_many_many = [
        'Supports' => Team::class,
    ];
}
class TeamSupporter extends DataObject
{
    private static $db = [
        'SomeData' => 'Varchar(10)',
    ];

    private static $has_one = [
        'Team' => Team::class,
        'Supporter' => Supporter::class,
    ];
}
```

With the following code in `Team::getCMSFields` (that works properly when `SomeData` is defined as `$many_many_extraFields`), I am able to change it (I see the new value in the database) but the form field element is not properly populated (it is always empty):
```php
$config->getComponent>ByType(GridFieldEditableColumns::class)->setDisplayFields([
    'SomeData' => 'A really useful field',
]);
```

With the following code, the form field is properly populated but it is turned readonly:
```php
$config->getComponent>ByType(GridFieldEditableColumns::class)->setDisplayFields([
    'TeamSupporter.SomeData' => 'A really useful field',
]);
```

Unfortunately I need to use an explicit through relation to be able to `populateDefaults()` the extra fields and add some calculated field. If anyone has a better idea I'm all ears.

**Comment by ntd** [comment]
I just opened a PR that allows me to use the first approach, i.e.:
```php
$config->getComponent>ByType(GridFieldEditableColumns::class)->setDisplayFields([
    'SomeData' => 'A really useful field',
]);
```

**Comment by ntd** [comment]
The recently merged #294, together with silverstripe/silverstripe-framework#9192, will allow to properly use dot syntax in this cases, e.g. the following code should work:
```php
$config->getComponent>ByType(GridFieldEditableColumns::class)->setDisplayFields([
    'TeamSupporter.SomeData' => 'A really useful field',
]);
```
Although not released, one can leverage `cweagans/composer-patches` to apply unmerged changes, e.g. by having a variant of the following text in your `composer.json`:
```json
{
  ...
  "require": {
    ...
    "cweagans/composer-patches": "^1.0"
  },
  "extra": {
    "patches": {
      "silverstripe/framework": {
        "Support dot syntax in ManyManyThroughList": "https://github.com/silverstripe/silverstripe-framework/pull/9192.patch"
      }
    }
  }
}
```

**Comment by ntd** [comment]
silverstripe/silverstripe-framework#9192 has been recently merged, so I consider this issue solved.
